### ISSUE TITLE
Style/BlockDelimiters (semantic) false positive when used in `case`

### ISSUE THREAD

**Issue by tylerhunt** [rubocop/rubocop#13720]
## Expected behavior

When using:

```yaml
Style/BlockDelimiters:
  EnforcedStyle: semantic
```

The following code should be valid:

```ruby
case hash.detect { |_, value| value }
in [key, _] then key
end
```

## Actual behavior

Fails with the following offense:

```
block_delimiters_false_positive.rb:3:18: C: [Correctable] Style/BlockDelimiters: Prefer do...end over {...} for procedural blocks.
case hash.detect { |_, value| value }
                 ^
```

## Steps to reproduce the problem

Configure Rubocop as above and use the example shown.

## RuboCop version

```
1.70.0 (using Parser 3.3.7.0, rubocop-ast 1.37.0, analyzing as Ruby 3.3, running on ruby 3.3.4) [arm64-darwin23]
  - rubocop-capybara 2.21.0
  - rubocop-rails 2.25.1
  - rubocop-rspec 3.0.3
```
